File: Server/src/main/java/com/codingfairy/bl/service/ActivenessService.java
package com.codingfairy.bl.service;

import com.codingfairy.bl.vo.ActivenessVo;
import com.codingfairy.exception.ParamException;
import com.codingfairy.exception.ServerException;
import com.codingfairy.utils.enums.QueryThreshold;
import com.codingfairy.web.json.Tuple;
import com.codingfairy.web.json.analysis.element.CustomerActivenessElement;

import java.util.List;

/**
 * Created by cuihao on 2017-05-16.
 * Customer activeness service
 */
public interface ActivenessService {

    List<ActivenessVo> getActivenessStat(int startHour, QueryThreshold threshold, int startRange, String probeId) throws ParamException;

    List<Tuple<String,Number>> findByHourAndProbe(int hour, String probeId);

    ActivenessVo findById(int id) throws ServerException;

    ActivenessVo save(CustomerActivenessElement activenessElement);
}

File: StatDisplay/src/services/activenessService.js
/**
 * Created by yyy on 2017/5/26.
 * to provide activeness data
 */

import {request} from "../utils/request";

export async function getActiveness(params) {
  return request(`/activeness/stat`,{
    method: `POST`,
    body: JSON.stringify({
      probeId: params.probeId,
      startHour: params.startHour,
      startRange: params.startRange,
      threshold: params.threshold
    })
  })
}

export async function getActivenessDetail(params) {
  return request(`/activeness/detail`,{
    method: `POST`,
    body: JSON.stringify({
      hour: params.hour,
      probeId: params.probeId
    })
  })
}


